novawallet/Common/Services/WalletConnect/WalletConnectService.swift

```diff
@@ -459,6 +459,18 @@ private final class DefaultWebSocket: WebSocketConnecting {
         startWebsocket()
     }
 
+    private func restart() {
+        mutex.lock()
+
+        defer {
+            mutex.unlock()
+        }
+
+        connectionState = .connecting
+
+        startWebsocket()
```

Comment:
I would call the method `protectedForceRestart()` and also call `stopWebsocket()` before starting new one